### DISCUSSION TITLE
Cutom PUBLICIP_HTTP_PROVIDERS  & PUBLICIPV4_HTTP_PROVIDERS

### DISCUSSION THREAD

**Discussion by D3an1el** [qdm12/ddns-updater#676] [Q&A] [answered]
Is it possible to provide a custom url  for  PUBLICIP_HTTP_PROVIDERS & PUBLICIPV4_HTTP_PROVIDERS address? as i currently run one with nginx.

**Comment by qdm12** [comment]
With `url:https://ipinfo.io/ip` for example it should work.

**Comment by D3an1el** [comment]
> With `url:https://ipinfo.io/ip` for a example it should work.
Tried that

i have tried with command & envfile in systemd

and also 

```
ExecStart=/opt/ddns_updater/ddns-updater --listening-address :8090 --health-server-address=192.168.1.148:9065 --publicip_http_providers=url:https://ipinfo.io/ip
``` 

env file

```
PUBLICIP_HTTP_PROVIDERS="url:https://ipinfo.io/ip"
PUBLICIPV4_HTTP_PROVIDERS="url:https://ipinfo.io/ip"
```

```
ExecStart=/opt/ddns_updater/ddns-updater --listening-address :8090 --health-server-address=192.168.1.148:9065 ${ENV}

```
edit:

error  in logs

```
ERROR settings validation: public ip settings: HTTP IP providers: value is not one of the possible choices: url:https://ipinfo.io/ip
```

**Reply by qdm12** [reply, marked as answer]
Actually, this was fixed a month ago in commit https://github.com/qdm12/ddns-updater/commit/877531c3d943418878acf2aead2a1e8531c041cf
I would guess you are running version 2.6.0 which doesn't have the fix unfortunately.
You would need to build the (still unreleased) last version, which you can do by:

1. [Install Go](https://go.dev/doc/install)
2. Run `go install github.com/qdm12/ddns-updater/cmd/updater@master` and this will install the statically compiled binary program in your path (I think /go/bin/updater by default). Compilation takes about 20 seconds.

Or wait for a v2.7.0 release if this isn't pressing 😉
